### PR TITLE
Ranker definition

### PR DESCRIPTION
**Pull request by moritzschaefer**
please review and merge

### CODE REVIEW

File: test/test_crawler.py [hunk 1]
```diff
@@ -1,11 +1,36 @@
-from tsg.crawler import crawl_site, crawl_urls, crawl_journal
+from tsg.crawler import crawl_site, crawl_urls, crawl_journal_url, crawl_journal_subsites
 import os
 import requests
 from tsg.config import RAW_DIR
 
+def test_crawl_journal():
+    urls = ['http://dblp.uni-trier.de/db/journals/ij3dim/ij3dim1.html',
```
Comment (moritzschaefer) [inline test/test_crawler.py]: that's not cool! but let it go if it works :)

File: test/test_ranker.py [hunk 2]
```diff
@@ -1,52 +1,123 @@
...
+def mock_dictionary_create() :
+	line1 = 'term c7c1d354-4b85-438b-bb2e-89350e40e33f:3.3322237271982384,15da4df3-9ef1-4e1a-b0ba-f93bf05a25d0:3.3205763030575843,7dd5a186-1dfe-4be6-be0b-ded65e8067c9:3.3205763030575843\n'
+	line2 = 'to c7c1d354-4b85-438b-bb2e-89350e40e33f:3.3205763030575843,15da4df3-9ef1-4e1a-b0ba-f93bf05a25d0:3.3322237271982384,7dd5a186-1dfe-4be6-be0b-ded65e8067c9:3.3205763030575843\n'
+	line3 = 'evaluate c7c1d354-4b85-438b-bb2e-89350e40e33f:3.3205763030575843,15da4df3-9ef1-4e1a-b0ba-f93bf05a25d0:3.3205763030575843,7dd5a186-1dfe-4be6-be0b-ded65e8067c9:3.3322237271982384\n'
+	mock_filename = DATA_DIR + 'mock_dictionary.dat'
+
+	with open(mock_filename,'w') as f:
+		f.write(line1)
+		f.write(line2)
+		f.write(line3)
```
Comment (moritzschaefer) [inline test/test_ranker.py]: @before(mock_dictionary_create)

File: test/test_ranker.py [hunk 3]
```diff
@@ -1,52 +1,123 @@
...
+	term3 = parts[2]
+	term3_list = get_dictionary_term_list(term3, mock_filename)
+	term3_query_weight = calculate_query_term_weight(term3,query,mock_filename)
+
+	Score = { } # Holds tf-idf value for each doc- query term pair
+	Length = { } # Holds numbers of docs for each term
+
+	for key,value in term1_list.items():
+		if key in Score:
+			Score[key] += float(value)*float(term1_query_weight)
```
Comment (moritzschaefer) [inline test/test_ranker.py]: lowercase this!

File: test/test_ranker.py [hunk 4]
```diff
@@ -1,52 +1,123 @@
...
+		else:
+			Score[key] = float(value)*float(term1_query_weight)
+
+		if key in Length:
+			Length[key] += math.pow(float(Score[key]),float(2))
+		else:
+			Length[key] = math.pow(float(Score[key]),float(2))
+
+	for key,value in term2_list.items():
```
Comment (moritzschaefer) [inline test/test_ranker.py]: key,<space>value

File: tsg/crawler/base.py [hunk 5]
```diff
@@ -21,23 +22,37 @@ def crawl_site(url, category):
         return
 
     webpage = get_site(url)
-    with open(doc_path, 'w') as f:
-        f.write(webpage.text)
+    if webpage.status_code != 404:
```
Comment (moritzschaefer) [inline tsg/crawler/base.py]: webpage.status_code == 200

File: tsg/crawler/base.py [hunk 6]
```diff
@@ -21,23 +22,37 @@ def crawl_site(url, category):
         return
 
     webpage = get_site(url)
```
Comment (moritzschaefer) [inline tsg/crawler/base.py]: try:
  webpage = get_site(url)
except uiaeuiae

else:
   save

File: tsg/crawler/downloader.py [hunk 7]
```diff
@@ -26,6 +26,9 @@ def get_site(url):
             logging.warn('There was a problem getting URL {}: Status code: {}'.
                          format(url, result.status_code))
 
+            if result.status_code == 404:
+                return result
```
Comment (moritzschaefer) [inline tsg/crawler/downloader.py]: raise lala

File: tsg/ranker/base.py [hunk 8]
```diff
@@ -1,9 +1,49 @@
...
+				Scores[key] += float(value)*float(query_term_weight)
+			else:
+				Scores[key] = float(value)*float(query_term_weight)
+
+			if key in Length:
+				Length[key] += math.pow(float(Scores[key]),float(2))
+			else:
+				Length[key] = math.pow(float(Scores[key]),float(2))
+
+	for key in Scores:
```
Comment (moritzschaefer) [inline tsg/ranker/base.py]: return something?

File: test/test_ranker.py (hunk 4 above)
Comment (brianignacio5) [inline test/test_ranker.py]: I don't understand what to do you mean here.

Comment (moritzschaefer) [inline test/test_ranker.py]: Ah wrong formatted from me sorry. After a ',' there is always a space.
key,value -> key, value
just a detail :)

File: test/test_ranker.py [hunk 9]
```diff
@@ -1,123 +1,54 @@
...
+	except OSError:
+		pass
+	assert not os.path.exists(mock_filename)
 
 	with open(mock_filename,'w') as f:
 		f.write(line1)
 		f.write(line2)
 		f.write(line3)
 
+mock_filename = DATA_DIR + 'mock_dictionary.dat'
```
Comment (moritzschaefer) [inline test/test_ranker.py]: Hmm this is not it.. sorry :)
http://nose.readthedocs.org/en/latest/writing_tests.html

Search in the page for 'with_setup'.
This is what you are looking for:

@with_setup(setup_func, teardown_func)
def func(arg):
    assert something_about(arg)

**Pull request merged**